app/xvinfo/xvinfo.man
.\" $XdotOrg: $
.\" $XFree86: xc/programs/xvinfo/xvinfo.man,v 1.2 2001/01/24 00:06:51 dawes Exp $
.\"
.TH xvinfo __appmansuffix__ __vendorversion__
.SH NAME
xvinfo - Print out X-Video extension adaptor information
.SH SYNOPSIS
.B "xvinfo"
[-display \fIdisplayname\fP] 
.SH DESCRIPTION
.PP
.B xvinfo
prints out the capabilities of any video adaptors associated
with the display that are accessible through the X-Video extension. 
.SH OPTIONS
.PP
.TP 8
.B "-display \fIdisplay\fP"
This argument allows you to specify the server to query; see \fIX(__miscmansuffix__)\fP.
.TP 8
.B "-short"
Output less details, to reduce the amount of text.
.SH ENVIRONMENT
.PP
.TP 8
.B DISPLAY
This variable may be used to specify the server to query.

.SH SEE ALSO
X(__miscmansuffix__), xdpyinfo(__appmansuffix__), xwininfo(__appmansuffix__),
xdriinfo(__appmansuffix__), glxinfo(__appmansuffix__), xprop(__appmansuffix__)
.SH AUTHORS
Mark Vojkovich


xserver/ChangeLog
commit 439c58849304907900e4dc7429aedb0192749c02
Author: Keith Packard <keithp@keithp.com>
Date:   Sun Oct 11 22:03:32 2009 -0700

    Bump version to 1.6.5
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 7f2f71da5125ae01a8279a4886090a711100d671
Author: Peter Hutterer <peter.hutterer@who-t.net>
Date:   Mon Sep 28 14:18:45 2009 +1000

    ephyr: if -parent is given, check for a trailing -screen. (#24144)
    
    If -parent is given, don't open up a new window if -screen is given as well.
    The commandline option -screen allows to set the depth of the embedded
    Xephry instance, even though width and height are autoscaled on -parent.
    
    This patch checks for a -screen parameter after -parent and - if one is
    found - delays initializing the screen. The parent window id is stored
    temporarily but re-set after a -screen argument.
    The following command is thus valid:
    
    Xephyr -parent 1234 -screen 640x480@8 -screen 1024x768
    
    It embeds the first 8-bit screen into window 1234 and opens up a new window
    for the second screen. Multiple parent arguments are possible, the screens
    are embedded in-order.
    
    X.Org Bug 24144 <http://bugs.freedesktop.org/show_bug.cgi?id=24144>
    
    Tested-by: Vic Lee
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
    (cherry picked from commit 19be992d9dc542b61fa3f4fd32a09071c9e64880)
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit c07b2368a7dedb66455eea313c06859afa2e1a9c
Author: Keith Packard <keithp@keithp.com>
Date:   Thu Oct 1 15:24:58 2009 -0700

    Bump to version 1.6.4.901
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit fc31f76b6ff735e4a4006c5454c4d496ee8b154a
Author: Keith Packard <keithp@keithp.com>
Date:   Wed Sep 30 11:40:19 2009 -0700

    Re-fix DGA removal.
    
    Removing DGA ended up breaking any drivers calling into the old
    xf86DiDGAInit function as it tried to see if DGA was already enabled
    and ended up crashing if the VT wasn't completely initialized. Oops.
    
    Also, if the driver initializes DGA itself, have the DiDGA
    initialization overwrite that information as the DiDGA code will call
    ReInit on mode detect.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
    (cherry picked from commit db98b26ee145f70e732e2cf4a6ac3de77fdf4adc)

commit 8cf659f3a5914369a2137ac17b689e5a9fe9ca27
Author: Keith Packard <keithp@keithp.com>
Date:   Sun Sep 27 19:21:45 2009 -0700

    Bump to version 1.6.4
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit a26fd1a6d61507b3e69a04d6f6c192a6ec363c5c
Author: Peter Hutterer <peter.hutterer@who-t.net>
Date:   Wed Sep 16 15:46:55 2009 +1000

    Don't send events through the master if the device has SendCoreEvents off.
    
    In server 1.6, all devices are attached to the master device (VCP or VCK).
    Sending an event through the master device means the device is sending core
    events. If a device is configured as SendCoreEvents, just send through the
    device, not through the master.
    
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

commit 507e57381fea6334f7dc8da6925e53d2c76fddcb
Author: Keith Packard <keithp@keithp.com>
Date:   Fri Sep 18 21:12:17 2009 -0700

    xfree86/modes: Remove all framebuffer support from DGA
    
    This removes all rendering and mapping code from xf86DiDGA, leaving
    just mode setting and raw input device access. The mapping code didn't
    have the offset within /dev/mem for the frame buffer and the pixmap
    support assumed that the framebuffer was never reallocated.
    (cherry picked from 0b7c6c728c2e2d8433a188315cc591308a89cd85)
    
    Signed-off-by: Keith Packard <keithp@keithp.com>
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

commit c3d182a47902d02f9e64f933b8565ae336f73f54
Author: Rémi Cardona <remi@gentoo.org>
Date:   Mon Sep 14 17:09:59 2009 +0200

    dix: append "built-ins" to the font path in SetDefaultFontPath
    
    49b93df8a3002db7196aa3fc1fd8dca1c12a55d6 made the hard dependency on
    a "fixed" font go away but only Xorg could use the built-ins fonts by
    default.
    
    With this commit, all DDXs get "built-ins" appended to their FontPath, not
    just Xorg.
    
    Tested with Xorg, Xvfb and Xnest.
    (cherry picked from commit f56cbe1ef24415d0142b9a7d0ab0a031069ccb52)
    
    Signed-off-by: Rémi Cardona <remi@gentoo.org>
    Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
    Tested-by: Jon TURNEY <jon.turney@dronecode.org.uk>
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 9bc4a69040493e589a3811ca5e085e323438996d
Author: Daniel Stone <daniel@fooishbar.org>
Date:   Wed Sep 9 15:48:33 2009 +1000

    fbdevhw: Test for graphics:fb%d as well as graphics/fb%d
    
    Apparently the kernel can't decide on an API to expose to userspace, so
    let's just try both in the hope that one will work.
    
    Signed-off-by: Daniel Stone <daniel@fooishbar.org>
    Acked-by: Michel Dänzer <michel@daenzer.net>
    (cherry picked from commit f4350c66b493d63fa06be87caa958d7033232ea4)

commit d78669c92d751464bc3c8f79f2aa21ca0951e3a9
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Sep 22 12:09:48 2009 -0700

    render: return the supported version rather than just passing the proto's version
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit f5d1da499c3d64f0a4ac217eb19911b8e35cb49b
Author: Michel Dänzer <daenzer@vmware.com>
Date:   Tue Jun 23 16:45:39 2009 +0200

    dri2: Don't crash if pPriv is NULL.
    (cherry picked from commit df597709d71f47b8516e27c6fb1bfffd59de5e48)
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 468787bdd2c34f9edf46da76b6357034c216fa7b
Author: Richard Hughes <richard@hughsie.com>
Date:   Fri Aug 14 11:44:35 2009 +0100


app/xvinfo/xvinfo.c
/* $XFree86: xc/programs/xvinfo/xvinfo.c,v 1.7 2001/10/28 03:34:43 tsi Exp $ */

#include <X11/X.h>
#include <X11/Xlib.h>
#include <X11/extensions/Xvlib.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

static void
PrintUsage(void)
{
   fprintf(stderr, "Usage:  xvinfo [-display host:dpy] [-short]\n");
   exit(0);
}

int main(int argc, char *argv[])
{
    Display *dpy;
    unsigned int ver, rev, eventB, reqB, errorB; 
    int i, j, k, n; 
    unsigned int nencode, nadaptors;
    int nscreens, nattr, numImages;
    XvAdaptorInfo *ainfo;
    XvAttribute *attributes;
    XvEncodingInfo *encodings;
    XvFormat *format;
    XvImageFormatValues *formats;
    char * disname = NULL;
    char shortmode = 0;

    if((argc > 4))
       PrintUsage();

    if(argc != 1) {
       for ( i = 1; i < argc; i ++ ) {
           if(! strcmp(argv[i], "-display")) {
	       disname = argv[i + 1];
	       i ++;
	   } else if(! strcmp(argv[i], "-short") )
	       shortmode = 1;
	     else {
	       PrintUsage();
	     }
       }
    }


    if(!(dpy = XOpenDisplay(disname))) 
    {
	fprintf(stderr, "xvinfo:  Unable to open display %s\n",
                       (disname != NULL) ? disname : XDisplayName(NULL));
	exit(-1);
    }

    if((Success != XvQueryExtension(dpy, &ver, &rev, &reqB, &eventB, &errorB))) 
    {
	fprintf(stderr, "xvinfo: No X-Video Extension on %s\n",
                        (disname != NULL) ? disname : XDisplayName(NULL));
	exit(0);
    } 
    else
    {
	fprintf(stdout, "X-Video Extension version %i.%i\n", ver, rev);
    }

    nscreens = ScreenCount(dpy);

    for(i = 0; i < nscreens; i++) {
  	fprintf(stdout, "screen #%i\n", i);
	XvQueryAdaptors(dpy, RootWindow(dpy, i), &nadaptors, &ainfo);

	if(!nadaptors) {
	    fprintf(stdout, " no adaptors present\n");
	    continue;
	} 

	for(j = 0; j < nadaptors; j++) {
	    fprintf(stdout, "  Adaptor #%i: \"%s\"\n", j, ainfo[j].name);
	    fprintf(stdout, "    number of ports: %li\n", ainfo[j].num_ports);
	    fprintf(stdout, "    port base: %li\n", ainfo[j].base_id);
	    fprintf(stdout, "    operations supported: ");
	    switch(ainfo[j].type & (XvInputMask | XvOutputMask)) {
	    case XvInputMask:
	     if(ainfo[j].type & XvVideoMask) 
		fprintf(stdout, "PutVideo ");
	     if(ainfo[j].type & XvStillMask) 
		fprintf(stdout, "PutStill ");
	     if(ainfo[j].type & XvImageMask) 
		fprintf(stdout, "PutImage ");
	     break;
	    case XvOutputMask:
	     if(ainfo[j].type & XvVideoMask) 
		fprintf(stdout, "GetVideo ");
	     if(ainfo[j].type & XvStillMask) 
		fprintf(stdout, "GetStill ");
	     break;
	    default:
		fprintf(stdout, "none ");
		break;
	    }
	    fprintf(stdout, "\n");

	    format = ainfo[j].formats;

	    if ( ! shortmode ) {
		    fprintf(stdout, "    supported visuals:\n");
		    for(k = 0; k < ainfo[j].num_formats; k++, format++) {
		         fprintf(stdout, "      depth %i, visualID 0x%2lx\n",
				 format->depth, format->visual_id);
		    }
	    }

	    attributes = XvQueryPortAttributes(dpy, ainfo[j].base_id, &nattr);

	    if(attributes && nattr) {
		fprintf(stdout, "    number of attributes: %i\n", nattr);
		
		for(k = 0; k < nattr; k++) {
		    fprintf(stdout, "      \"%s\" (range %i to %i)\n",
					attributes[k].name, 
					attributes[k].min_value,
					attributes[k].max_value);

		    if(attributes[k].flags & XvSettable)
			if ( ! shortmode )
		    	  fprintf(stdout, "              client settable attribute\n");
		        else fprintf(stdout, "              settable");

		    if(attributes[k].flags & XvGettable) {
			Atom the_atom;
			int value;
			
			if ( ! shortmode )
		    	  fprintf(stdout, "              client gettable attribute");
			else fprintf(stdout, ", gettable");

			the_atom = XInternAtom(dpy, attributes[k].name, True);

			if(the_atom != None){
			   if((Success == XvGetPortAttribute(dpy, 
					ainfo[j].base_id, the_atom, &value)))
		    	       fprintf(stdout, " (current value is %i)", value);
			}
			fprintf(stdout, "\n");
		    }
		    else if (shortmode) 
			fprintf(stdout, "\n");

		}
		XFree(attributes);
	    } else {
		fprintf(stdout, "    no port attributes defined\n");
	    }

	    XvQueryEncodings(dpy, ainfo[j].base_id,
						&nencode, &encodings);

	    if(encodings && nencode) {
		int ImageEncodings = 0;

		for(n = 0; n < nencode; n++) {
		    if(!strcmp(encodings[n].name, "XV_IMAGE"))
			ImageEncodings++;
		}

		if(nencode - ImageEncodings) {
		    fprintf(stdout, "    number of encodings: %i\n", 
						nencode - ImageEncodings);

		    for(n = 0; n < nencode; n++) {
		    	if(strcmp(encodings[n].name, "XV_IMAGE")) {
			    fprintf(stdout, "      encoding ID #%li: \"%s\"\n",
					encodings[n].encoding_id,
					encodings[n].name);
			    fprintf(stdout, "        size: %li x %li\n",
					encodings[n].width,
					encodings[n].height);
			    fprintf(stdout, "        rate: %f\n",
					(float)encodings[n].rate.numerator/
					(float)encodings[n].rate.denominator);
			}
		    }
		}

		if(ImageEncodings && (ainfo[j].type & XvImageMask)) {
		   char imageName[5];

		   for(n = 0; n < nencode; n++) {
		      if(!strcmp(encodings[n].name, "XV_IMAGE")) {
			fprintf(stdout, 
				"    maximum XvImage size: %li x %li\n",	
				encodings[n].width, encodings[n].height);
			break;
		      }
		   }

                   formats = XvListImageFormats(
				dpy, ainfo[j].base_id, &numImages);

		   fprintf(stdout, "    Number of image formats: %i\n",
						numImages);

		   for(n = 0; n < numImages; n++) {
			sprintf(imageName, "%c%c%c%c", formats[n].id & 0xff,
				(formats[n].id >> 8) & 0xff,
				(formats[n].id >> 16) & 0xff,
				(formats[n].id >> 24) & 0xff);
			fprintf(stdout, "      id: 0x%x", formats[n].id);
			if(isprint(imageName[0]) && isprint(imageName[1]) &&
			  isprint(imageName[2]) && isprint(imageName[3])) 
			{
			  fprintf(stdout, " (%s)\n", imageName);
			} else {
			  fprintf(stdout, "\n");
			}
			if ( ! shortmode ) {
			    fprintf(stdout, "        guid: ");
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[0]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[1]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[2]);
			    fprintf(stdout, "%02x-", (unsigned char) 
						formats[n].guid[3]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[4]);
			    fprintf(stdout, "%02x-", (unsigned char) 
						formats[n].guid[5]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[6]);
			    fprintf(stdout, "%02x-", (unsigned char) 
						formats[n].guid[7]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[8]);
			    fprintf(stdout, "%02x-", (unsigned char) 
						formats[n].guid[9]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[10]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[11]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[12]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[13]);
			    fprintf(stdout, "%02x", (unsigned char) 
						formats[n].guid[14]);
			    fprintf(stdout, "%02x\n", (unsigned char) 
						formats[n].guid[15]);

			    fprintf(stdout, "        bits per pixel: %i\n",
					formats[n].bits_per_pixel);
			    fprintf(stdout, "        number of planes: %i\n",
					formats[n].num_planes);
			    fprintf(stdout, "        type: %s (%s)\n", 
			      (formats[n].type == XvRGB) ? "RGB" : "YUV",
			      (formats[n].format == XvPacked) ? "packed" : "planar");

			    if(formats[n].type == XvRGB) {
			        fprintf(stdout, "        depth: %i\n", 
                                        formats[n].depth);

			        fprintf(stdout, "        red, green, blue masks: " 
					"0x%x, 0x%x, 0x%x\n", 
					formats[n].red_mask,
					formats[n].green_mask,
					formats[n].blue_mask);
			    } else {

			    } 
			}
			
	
		   }
		   if(formats) XFree(formats);
		}

		XvFreeEncodingInfo(encodings);
	    }
	    
	}

	XvFreeAdaptorInfo(ainfo);
    }
    return 1;
}
